src/components/add/CardNumberInput/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import { Card } from 'src/types/card'
+
+import { BasicInput, Input, InputBox, InputTitle, InputWarning } from '$components/common/Input'
+import { checkLength, checkNumberString } from '$utils/validate'
+
+interface CardNumberInputProps {
+  cardNumber: Card['number']
+  setCardNumber: (cardNumber: Card['number']) => void
+}
+
+function CardNumberInput({ cardNumber, setCardNumber }: CardNumberInputProps) {
+  const ref1 = useRef<HTMLInputElement>(null)
+  const ref2 = useRef<HTMLInputElement>(null)
+  const ref3 = useRef<HTMLInputElement>(null)
+  const ref4 = useRef<HTMLInputElement>(null)
+  const [number1, setNumber1] = useState('')
+  const [number2, setNumber2] = useState('')
+  const [number3, setNumber3] = useState('')
+  const [number4, setNumber4] = useState('')
```

Comment:
각각의 input을 별도로 선언하지 않고 하나의 객체로 선언해서 관리하는 방법을 사용해도 좋을거 같은데 어떻게 생각하시나요?

```suggestion
  const [cardNumber, setCardNumber] = useState({
    first: '',
    second: '',
    third: '',
    fourth: '',
  });
```

이런 식으로 관리하면 아래처럼 handler함수를 4개 만들 필요 없이 하나로 전부 관리할 수 있을거 같아요!

Reply:
좋은 아이디어 감사합니다!  9c8e3a4 에서 수정했습니다.
password 도 있긴한데,,, 일단 card number 에만 적용해봤습니다.